platform/engineering/frontend/eslint/eslint-testing.md

```diff
@@ -0,0 +1,57 @@
+# ESLint Rule Changes and Testing
+
+**Author:** Jhonny Gonzalez
+**Last Updated:** February 21, 2020
+**eslint version:** 4.18.2
+
+## Overview
+
+This is an initial process to test and deploy new rules for ESLint.
+
+## Objective
+
+To improve the code coverage and code quality in the repo by enforcing new ESLint rules. This also will help all VFS teams to write better code.
+
+## Background
+
+We installed ESLint a couple of years ago in the VA repo and hasn't been updated nor improved since then.
+
+Now, that VFS teams will have an increased ownership on their pull request reviews, this will help them getting a better judgement as well as facilitate the review process.
+
+# Specifics
+
+## ESLintin Procedure
+
+The ESLinting rules in CircleCI will be used as an experimental staging area as we explore new rules and ways to communicate any issues that are discovered.
+
+## Code Location
+
+### ESLint main file
+
+`.eslintrc` => located at the root folder.
+
+### ESLint Testing file
+
+`circle.eslint.json` => rules under review will be located here
+
+## Plugin/New Rules Audit
+
+Before contemplating the idea of adding a new plugin with rules, it is important to perform an audit. This audit will help us determine if the current repo is in compliance with each rule. All outstanding errors from rules must be cleared (disabled or fixed) before going into the testing stage.
+
+VFS teams are responsible for fixing the errors generated by the rule under review. Once all the errors have been cleared, we can proceed by adding the rule(s) to the testing stage file (`circle.eslint.json`)
+
+## Rules acceptance
+
+All rules under review will always be considered to become enforced. However, based on feedback or complexity they can become disabled and not enforced.
```

Comment:
It might be clearer to explain this in terms of the rule's lifecycle. So like:
1. Rule X is considered
2. Rule X is added to the unenforced ruleset
3. Rule X is test run in the unenforced ruleset for two weeks
4. Rule X is added to the enforced ruleset
    - Unless there are unresolved errors
    - Unless there has been significant pushback against the rule and VSP decides to not enforce it

Reply:
Wow, I like this